src/main/pages/che-7/installation-guide/proc_gerenating-self-signed-certificates.adoc

```diff
@@ -0,0 +1,130 @@
+[id="gerenating-self-signed-certificates_{context}"]
+= Generating self-signed TLS certificates
+
+This section describes how to prepare self-signed TLS certificates to use with {prod-short} on different platforms.
+
+
+[discrete]
+== Prerequisites
+
+* Find out expected domain name where {prod-short} deployment is planned
+
++
+[subs="+quotes"]
+----
+# default for minishift
+$( minishift ip ).nip.io
+
+# default for minikube
+$( minikube ip ).nip.io
+
+# default for crc
+apps-crc.testing
+----
+
+* Find out `openssl.cnf` file location on target machine
+
++
+[subs="+quotes"]
+----
+# Fedora 31
+/etc/pki/tls/openssl.cnf
+----
+
+[discrete]
+== Procedure
+
+. Set necessary environment variables
+
++
+[subs="+quotes"]
+----
+$ CA_CN=eclipse-che-signer
+$ DOMAIN=*.<expected.domain.com>
+$ OPENSSL_CNF=<path_to_openssl.cnf>
+----
+
++
+[subs="+quotes"]
+----
+# Example with minikube on Fedora 31
+$ CA_CN=eclipse-che-signer
+$ DOMAIN=*.$( minikube ip ).nip.io
+$ OPENSSL_CNF=/etc/pki/tls/openssl.cnf
+
+# Example with crc on OSX
+$ export CA_CN=eclipse-che-signer
+$ export DOMAIN=*.apps-crc.testing
+$ export OPENSSL_CNF=/System/Library/OpenSSL/openssl.cnf
+----
+
+. Generate root CA key. Add `-des3` parameter if passphrase is needed
+
++
+[subs="+quotes"]
+----
+$ openssl genrsa -out rootCA.key 4096
+----
+
+
+. Generate root certificate.
+
++
+[subs="+quotes"]
+----
+$ openssl req -x509 \
+  -new -nodes \
+  -key rootCA.key \
+  -sha256 \
+  -days 1024 \
+  -out rootCA.crt \
+  -subj /CN=${CA_CN} \
+  -reqexts SAN \
+  -extensions SAN \
+  -config <(cat ${OPENSSL_CNF} \
+      <(printf '[SAN]\nbasicConstraints=critical, CA:TRUE\nkeyUsage=keyCertSign, cRLSign, digitalSignature, keyEncipherment'))
+
+----
+
+
+. Generate domain key.
+
++
+[subs="+quotes"]
+----
+$ openssl genrsa -out domain.key 2048
+----
+
+. Generate domain csr.
+
++
+[subs="+quotes"]
+----
+$ openssl req -new -sha256 \
+    -key domain.key \
+    -subj "/O=EclipseChe/CN=${DOMAIN}" \
+    -reqexts SAN \
+    -config <(cat ${OPENSSL_CNF} \
+        <(printf "\n[SAN]\nsubjectAltName=DNS:${DOMAIN}\nbasicConstraints=critical, CA:FALSE\nkeyUsage=keyCertSign, digitalSignature, keyEncipherment\nextendedKeyUsage=serverAuth")) \
+    -out domain.csr
+----
+
+
+. Generate domain certificate.
+
++
+[subs="+quotes"]
+----
+$ openssl x509 \
+-req \
+-extfile <(printf "subjectAltName=DNS:${DOMAIN}\nbasicConstraints=critical, CA:FALSE\nkeyUsage=keyCertSign,                       digitalSignature, keyEncipherment\nextendedKeyUsage=serverAuth") \
```

Comment:
```
-sha256 \
```
Reason - command generate SHA1 based crt file on OSX

Reply:
works fine on my Fedora 31. I've patched the docs with this.